fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBIndexOutput.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * FDBIndexOutput.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.directory;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.annotation.SpotBugsSuppressWarnings;
+import org.apache.lucene.store.DataInput;
+import org.apache.lucene.store.IndexOutput;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.zip.CRC32;
+
+/**
+ *
+ * Implementation of IndexOutput representing the writing of data
+ * in Lucene to a file.
+ *
+ * @see <a href="https://lucene.apache.org/core/7_6_0/core/org/apache/lucene/store/IndexOutput.html">IndexOutput</a>
+ */
+@API(API.Status.EXPERIMENTAL)
+public final class FDBIndexOutput extends IndexOutput {
+    private static final Logger LOG = LoggerFactory.getLogger(FDBIndexOutput.class);
+    private int currentSize = 0;
+    private ByteBuffer buffer;
+    private final String resourceDescription;
+    private final FDBDirectory fdbDirectory;
+    private final long blockSize;
+    private final CRC32 crc;
+    private final long id;
+    public static final ArrayBlockingQueue<ByteBuffer> BUFFERS;
+    public static final int POOL_SIZE = 100;
+
+    static {
+        BUFFERS = new ArrayBlockingQueue<>(POOL_SIZE);
+        for (int i = 0; i < POOL_SIZE; i++) {
+            BUFFERS.add(ByteBuffer.allocate(FDBDirectory.DEFAULT_BLOCK_SIZE));
+        }
+    }
+
+    /**
+     * Create an FDBIndexOutput given a resource description, name, and FDBDirectory.
+     *
+     * @param resourceDescription description of resource
+     * @param name name of resource
+     * @param fdbDirectory existing FDBDirectory
+     */
+    public FDBIndexOutput(String resourceDescription, String name, FDBDirectory fdbDirectory) {
+        super(resourceDescription, name);
+        LOG.trace("init() -> resource={}, name={}", resourceDescription, name);
+        this.resourceDescription = resourceDescription;
+        this.fdbDirectory = fdbDirectory;
+        blockSize = fdbDirectory.getBlockSize();
```

Comment:
What happens here if `fdbDirectory.getBlockSize()` is larger than `DEFAULT_BLOCK_SIZE`? Will that lead to errors down the line?

Reply:
Good ?...  Sounds like we need some tests.  :)